Common/Securities/SecurityExchangeHours.cs

```diff
@@ -335,5 +335,37 @@ private bool IsTimeAfterEarlyClose(DateTime localDateTime)
             TimeSpan earlyCloseTime;
             return _earlyCloses.TryGetValue(localDateTime.Date, out earlyCloseTime) && localDateTime.TimeOfDay >= earlyCloseTime;
         }
+
+        /// <summary>
+        /// Gets the previous trading day
+        /// </summary>
+        /// <param name="date">The date to start searching at</param>
+        /// <returns>The previous trading day</returns>
+        public DateTime GetPreviousTradingDay(DateTime date)
+        {
+            date = date.AddDays(-1);
+            while (!IsDateOpen(date))
```

Comment:
Might be nice to name the variables "localDate" for clarity in exchange timezone (i think?)